include/libp2p/basic/cancel.hpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+
+namespace libp2p {
+  /**
+   * Interface with destructor for `std::unique_ptr`.
+   */
+  class CancelDtor {
+   public:
+    virtual ~CancelDtor() = default;
```

Comment:
The only comment on the whole PR.
 I see no reason to make it constructable.
 If so, this could be an option:
```suggestion
    virtual ~CancelDtor() = 0;
```

Reply:
Thanks.

Definition (`CancelDtor::~CancelDtor() {}`) is [required for pure virtual destructor](https://en.cppreference.com/w/cpp/language/destructor#Pure_virtual_destructors).